compiler-rt/lib/scudo/standalone/string_utils.cpp

```diff
@@ -229,6 +229,15 @@ void ScopedString::append(const char *Format, ...) {
   va_end(Args);
 }
 
+size_t ScopedString::copyToBuffer(char *OutputBase, size_t OutputLength) {
+  if (OutputBase && OutputLength) {
+    const size_t Written = Min(length(), OutputLength - 1);
+    memcpy(OutputBase, data(), Written);
+    OutputBase[Written] = '\0';
+  }
+  return length() + 1;
```

Comment:
I think this should return the length of data copied?

Reply:
Here I tried to conform to existing APIs, they are a bit surprising, but it makes sense.

In combined.h, getStats uses that protocol, where the caller can provide a buffer, and if it is too small, the value gets truncated, and the caller knows it from the returned size. I reuse the same principle here.